File: src/main/java/org/luxons/sevenwonders/game/effects/EndGameEffect.java
package org.luxons.sevenwonders.game.effects;

import org.luxons.sevenwonders.game.api.Table;

public abstract class EndGameEffect implements Effect {

    @Override
    public void apply(Table table, int playerIndex) {
        // EndGameEffects don't do anything when applied to the board, they simply give more points in the end
    }
}

File: sw-engine/src/main/kotlin/org/luxons/sevenwonders/engine/effects/EndGameEffect.kt
package org.luxons.sevenwonders.engine.effects

import org.luxons.sevenwonders.engine.Player

internal abstract class EndGameEffect : Effect {

    // EndGameEffects don't do anything when applied to the board, they simply give more points in the end
    override fun applyTo(player: Player) = Unit
}
